src/bin/pg_autoctl/pgsql.c

```diff
@@ -1487,6 +1488,7 @@ pgsql_replication_slot_maintain(PGSQL *pgsql, NodeAddressArray *nodeArray)
 		"   FROM pg_replication_slots pgrs LEFT JOIN nodes USING(slot_name) "
 		"  WHERE nodes.slot_name IS NULL "
 		"    AND slot_name ~ '" REPLICATION_SLOT_NAME_PATTERN "' "
+		"    AND not active"
```

Comment:
This is necessary so that we don't impact the Postgres service when replacing the monitor. Dropping the replication slot while it's active doesn't work (ereport ERROR), so that we would need to stop streaming replication to drop it. I would rather keep things running as they are if possible.

About the confusion when nodes have a new id, I think it there would be only a small amount of time where the wrong slot would be used but then the system is going to fix itself when moving forward. It might be possible to think of a corner case where a node that is far behind gets assigned a slot that is further ahead, breaking its streaming abilities. Will think about that and how to approach it.

Reply:
Meanwhile and per review, we have added the capability to keep the node ids as with the previous monitor when doing `pg_autoctl enable monitor`, which allows re-using replication slots.